### DISCUSSION TITLE
Please help with "voxel size"; is it related to pixels?

### DISCUSSION THREAD

**Discussion by Aceyaa** [PMEAL/porespy#603] [Q&A] [answered]
Hi, I am trying to determine pore sizes using the local thickness function, but I could not figure out how many pixels are in each voxel in the outcome.

Using the example provided (https://porespy.org/examples/filters/tutorials/local_thickness.html?highlight=local%20thickness):

```python
R = 10
strel = ps.tools.ps_disk(R)
im_temp = spim.binary_opening(im, structure=strel)
fig, ax = plt.subplots()
ax.imshow(im_temp*2.0 + ~im);
```

The above indicated that R in calculating local thickness started as 10 pixels, and Rs appear to become smaller in pixel sizes for follow-up fittings. However, when graphing "x=psd.LogR", LogR is maxed at about 1.2 voxels. 

I read in other documents that voxels are scalar values and can have arbitrary units. So how do voxels relate to the image being analyzed? Do the voxels relate to pixels?  

Or, is there a way of correlating the size of the image I import for analysis with voxel size? I want to compare the pore size distribution of different samples. This won't work if different images have different voxel scales/sizes

Thank you!

**Comment by jgostick** [comment, marked as answer]
Firstly, I am very sorry for the late response to this.  I usually get email notifications but seemed to have missed this one.  

To answer your first question, the reason you're seeing a value of 1.2 pixels/voxels is that we take the log of the voxel radius so it's actually log(R), or to get the actual voxel radius back you do ``10**(log(R))``.  

Regarding your second point, the size of the voxels should be included with the tomography image you're analyzing.  It's usually something between 1 to 10 um per voxel.  Note that this number is the length of one side of a voxel.  Basically, if you are able to inscribe a sphere with a radius of 10 voxels, and your voxel size 5 um, then the actual sphere size is 50 um, or 0.000050 m.  You can input this into the Washburn equation to get capillary pressure if you want to plot a capillary pressure curve.

Hope that helps.

**Reply by OhmFF** [reply]
Dear sir,
What if the voxel is not perfectly cube (i.e., rectangular [x,y,z] of [10x10x20] nm)?
When using local thickness in 3-dimensional pore space, the multiplier for log(R) should be 10 or 20 nm? 

Thank you very much

**Reply by jgostick** [reply]
I'm afraid that our code is 100% hard coded to only accept equal voxel lenghts at this time.  There is a work around though...you can ``zoom`` the image in the large dimension to make the voxel sizes equal.  In the case you mention with 10 x 10 x 20, you'll do:

``` python
from scipy.ndimage import zoom
im = zoom(im, zoom=[1, 1, 2], order=0)
```

Then you're image will be twice as big in the direction, meaning that voxel lenghts are 10 nm instead of 20.  I suggest using order of 0, which is nearest neighbor.  Higher orders are much slower and require some processing afterward
